module/state_synchronization/requester/execution_data_requester.go

```diff
@@ -196,57 +181,12 @@ func New(
 	}
 	e.blockConsumer = blockConsumer
 
-	// notifies notificationConsumer when new ExecutionData blobs are available
-	// SetPostNotifier will notify executionDataNotifier AFTER e.blockConsumer.LastProcessedIndex is updated.
-	// Even though it doesn't guarantee to notify for every height at least once, the notificationConsumer is
-	// able to guarantee to process every height at least once, because the notificationConsumer finds new jobs
-	// using executionDataReader which finds new heights using e.blockConsumer.LastProcessedIndex
-	e.blockConsumer.SetPostNotifier(func(module.JobID) { executionDataNotifier.Notify() })
-
-	// jobqueue Jobs object tracks downloaded execution data by height. This is used by the
-	// notificationConsumer to get downloaded execution data from storage.
-	e.executionDataReader = jobs.NewExecutionDataReader(
-		e.execDataCache,
-		e.config.FetchTimeout,
-		// method to get highest consecutive height that has downloaded execution data. it is used
-		// here by the notification job consumer to discover new jobs.
-		// Note: we don't want to notify notificationConsumer for a block if it has not downloaded
-		// execution data yet.
-		func() (uint64, error) {
-			return e.blockConsumer.LastProcessedIndex(), nil
-		},
-	)
-
-	// notificationConsumer consumes `OnExecutionDataFetched` events, and ensures its consumer
-	// receives this event in consecutive block height order.
-	// It listens to events from `executionDataNotifier`, which is delivered when
-	// a block's execution data is downloaded and stored, and checks the `executionDataCache` to
-	// find if the next un-processed consecutive height is available.
-	// To know what's the height of the next un-processed consecutive height, it reads the latest
-	// consecutive height in `processedNotifications`. And it's persisted in storage to be crash-resistant.
-	// When a new consecutive height is available, it calls `processNotificationJob` to notify all the
-	// `e.consumers`.
-	// Note: the `e.consumers` will be guaranteed to receive at least one `OnExecutionDataFetched` event
-	// for each sealed block in consecutive block height order.
-	e.notificationConsumer, err = jobqueue.NewComponentConsumer(
```

Comment:
I simplified the requester by removing the notification consumer entirely. Instead, we are just calling `e.distributor.OnExecutionDataReceived()`. 

Note, the `OnExecutionDataReceived` used to be called with the execution data read from storage, but actually, no consumer actually make use the execution data, because the consumer will read execution data with their next unprocessed index to ensure data for all heights are processed.